Gems/Multiplayer/Code/Source/Components/NetworkDebugPlayerIdComponent.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Multiplayer/Components/NetworkDebugPlayerIdComponent.h>
+
+#include <Atom/RPI.Public/ViewportContextBus.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzFramework/Viewport/ViewportScreen.h>
+#include <AzNetworking/Framework/INetworking.h>
+
+namespace Multiplayer
+{
+    AZ_CVAR(bool, cl_debugPlayerIdRender, true, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Whether NetworkDebugPlayerIdComponent should render the player id and connection count on-screen.");
```

Comment:
Are you sure you want to default this to true?  I'd go with setting this to false in the gem so it isn't on by default for all projects, and setting it to true in our specific MPS Registry/game.setreg

Reply:
If it's off by default, it could be confusing to use this component. You find it in the component list, add it to an entity hoping to get extra info, and nothing happens. I don't know how discoverable the cvars are from the entity inspector that shows this component.

Would it be possible to add a boolean checkbox toggle that actually flips the cvar, on the inspector for this component?

Reply:
_"Would it be possible to add a boolean checkbox toggle that actually flips the cvar, on the inspector for this component?"_
Now that would be awesome. Like a component editor property that very clearly allows users to change a global setting. Is that possible? Is there an existing UX example of a component doing that?

Reply:
Good point, I wonder if this should be a single system component instead that monitors the cvar, instead of a component you need to remember to add to your players.

Reply:
No idea on how possible that would be. My guess is yes? Can't you, with the component property editor, mark some properties as not being variables on a component but instead the result of a function?

I haven't spent a lot of time working with the edit context, though, so I'm not totally positive.

Reply:
" I wonder if this should be a single system component instead that monitors the cvar, instead of a component you need to remember to add to your players."
I actually really like this idea. Sort of nice multiplayer-components handles the traffic, but this could instead be sending network packets to and from a system component. That system component could have a cvar controlled on the server side... server cvars can be marked for replication so clients know when it's toggled on and off; clients could listen in and receive the packets and render player id's when the cvar is enabled. The packet would include each player's id and currently world-space location. Would need to be careful to only update the players that haven't moved much. It also wouldn't be predictive, so you'd be seeing your own player id location in your past (jarring?). Sending packets directly would also be missing out on the implicit net-entity update filtration for hacker avoidance (ie: not sending other player position properties which the player cant currently see). However, this is for debugging anyways and devs would disable it for release.

Reply:
> No idea on how possible that would be. My guess is yes? Can't you, with the component property editor, mark some properties as not being variables on a component but instead the result of a function?
> 
> I haven't spent a lot of time working with the edit context, though, so I'm not totally positive.

I could link the component's edit-context to a cvar, but I'm more worried about the ux. Maybe just label it like... "GLOBAL SETTING: Enable Rendering". I that that would be a first... the cleanest way would be probably be @AMZN-alexpete suggestion and just remove this component completely and have a system component render all this.

Reply:
> It also wouldn't be predictive

Actually, that might not be true. I could send the network entity id instead of the position to the clients. The system component could use grab the entities by hand. That would well reduce net traffic too since you could send the player id once and forget it. Net Entity Id is persistent forever (I THINK)

Reply:
Rendering a player id above the player's head does seem like the sort of thing each player would be in charge of though, right? Rather than a system component I mean? I do think sending the total connection count would be nicer handled via system component though (for a later improvement).